### PR TITLE
fix(instance): Bump scaleway sdk go & replace instance volume server template type

### PR DESCRIPTION
**Pull request by Monitob**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -295,6 +295,8 @@ github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4/go.mod h1:
 github.com/rogpeppe/go-internal v1.3.0/go.mod h1:M8bDsm7K2OlrFYOpmOWEs/qY81heoFRclV5y23lUDJ4=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.7.0.20211109132617-29e472beb5cd h1:/jejlYKHj9/rSQbONniMDA2aZEDvGdDK0+1PFGT9b8Y=
 github.com/scaleway/scaleway-sdk-go v1.0.0-beta.7.0.20211109132617-29e472beb5cd/go.mod h1:fCa7OJZ/9DRTnOKmxvT6pn+LPWUptQAmHF/SBJUGEcg=
+github.com/scaleway/scaleway-sdk-go v1.0.0-beta.7.0.20211228142246-d7e9096b97a8 h1:FmxkX9nmraMmu7u7S53bk3dtm7U1H9AQ2ULo8/CRwAo=
```
Comment (remyleone) [inline go.sum:298]: go mod tidy

File: scaleway/helpers_instance.go [hunk 2]
```diff
@@ -230,11 +244,11 @@ func sanitizeVolumeMap(serverName string, volumes map[string]*instance.VolumeTem
 		// If a volume already got an ID it is passed as it to the API without specifying the volume type.
 		// TODO: Fix once instance accept volume type in the schema validation
 		case v.ID != "":
-			v = &instance.VolumeTemplate{ID: v.ID, Name: v.Name}
+			v = &instance.VolumeServerTemplate{ID: v.ID, Name: v.Name}
```
Comment (remyleone) [inline scaleway/helpers_instance.go:247]: put one argument per line to have something that is diff friendly

File: scaleway/helpers_instance.go [hunk 3]
```diff
@@ -230,11 +244,11 @@ func sanitizeVolumeMap(serverName string, volumes map[string]*instance.VolumeTem
...
 		// For the root volume (index 0) if the specified size is not 0 it is considered as a new volume
 		// It does not have yet a volume ID, it is passed to the API with only the size to be dynamically created by the API
 		case index == "0" && v.Size != 0:
-			v = &instance.VolumeTemplate{Size: v.Size}
+			v = &instance.VolumeServerTemplate{Size: v.Size}
```
Comment (remyleone) [inline scaleway/helpers_instance.go:251]: put one argument per line to have something that is diff friendly

File: scaleway/helpers_instance.go [hunk 4]
```diff
@@ -78,6 +78,20 @@ func orderVolumes(v map[string]*instance.Volume) []*instance.Volume {
 	return orderedVolumes
 }
 
+// orderServerVolumes return an ordered slice based on the volume map key "0", "1", "2",...
+func orderServerVolumes(v map[string]*instance.VolumeServer) []*instance.VolumeServer {
```
Comment (remyleone) [inline scaleway/helpers_instance.go:82]: Why is a new function needed?

File: scaleway/resource_instance_server.go [hunk 5]
```diff
@@ -302,27 +302,32 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
...
+				Project:    projectID,
 			}
 		}
 	} else {
 		if size, ok := d.GetOk("root_volume.0.size_in_gb"); ok {
-			req.Volumes["0"] = &instance.VolumeTemplate{
+			req.Volumes["0"] = &instance.VolumeServerTemplate{
 				Size:       scw.Size(uint64(size.(int)) * gb),
 				VolumeType: instance.VolumeVolumeTypeLSSD,
+				Project:    projectID,
```
Comment (remyleone) [inline scaleway/resource_instance_server.go:322]: Maybe a problem can come from adding project here.

File: scaleway/resource_instance_server.go [hunk 6]
```diff
@@ -302,27 +302,32 @@ func resourceScalewayInstanceServerCreate(ctx context.Context, d *schema.Resourc
...
 			}
 		} else {
 			// We add a local root volume if it is not already present
-			req.Volumes["0"] = &instance.VolumeTemplate{
+			req.Volumes["0"] = &instance.VolumeServerTemplate{
 				Name:       newRandomName("vol"),
 				VolumeType: instance.VolumeVolumeTypeLSSD,
 				Size:       serverType.VolumesConstraint.MinSize,
+				Project:    projectID,
```
Comment (remyleone) [inline scaleway/resource_instance_server.go:330]: Maybe a problem can come from adding project here.

File: scaleway/resource_instance_server.go [hunk 7]
```diff
@@ -649,12 +656,13 @@ func resourceScalewayInstanceServerUpdate(ctx context.Context, d *schema.Resourc
...
+	volumes := map[string]*instance.VolumeServerTemplate{}
 
 	if raw, ok := d.GetOk("additional_volume_ids"); d.HasChange("additional_volume_ids") && ok {
-		volumes["0"] = &instance.VolumeTemplate{
-			ID:   expandZonedID(d.Get("root_volume.0.volume_id")).ID,
-			Name: newRandomName("vol"), // name is ignored by the API, any name will work here
+		volumes["0"] = &instance.VolumeServerTemplate{
+			ID:      expandZonedID(d.Get("root_volume.0.volume_id")).ID,
+			Name:    newRandomName("vol"), // name is ignored by the API, any name will work here
+			Project: projectID,
```
Comment (remyleone) [inline scaleway/resource_instance_server.go:665]: Maybe a problem can come from adding project here.

File: scaleway/resource_instance_server.go [hunk 8]
```diff
@@ -676,9 +684,10 @@ func resourceScalewayInstanceServerUpdate(ctx context.Context, d *schema.Resourc
 					}
 				}
 			}
-			volumes[strconv.Itoa(i+1)] = &instance.VolumeTemplate{
-				ID:   expandZonedID(volumeID).ID,
-				Name: newRandomName("vol"), // name is ignored by the API, any name will work here
+			volumes[strconv.Itoa(i+1)] = &instance.VolumeServerTemplate{
+				ID:      expandZonedID(volumeID).ID,
+				Name:    newRandomName("vol"), // name is ignored by the API, any name will work here
+				Project: projectID,
```
Comment (remyleone) [inline scaleway/resource_instance_server.go:690]: Maybe a problem can come from adding project here.

File: scaleway/helpers_instance.go (hunk 4 above)
Comment (Monitob) [reply scaleway/helpers_instance.go:82]: This method is similar but manage a different type

File: scaleway/resource_instance_server.go (hunk 6 above)
Comment (Monitob) [reply scaleway/resource_instance_server.go]: ProjetID just removed

File: scaleway/resource_instance_server.go (hunk 7 above)
Comment (Monitob) [reply scaleway/resource_instance_server.go]: Yes it is, It was removed and test are OK

File: scaleway/helpers_instance.go (hunk 4 above)
Comment (remyleone) [inline scaleway/helpers_instance.go:82]: Rename to orderVolumeServer

Comment (Monitob) [reply scaleway/helpers_instance.go:82]: Maybe more appropriated here is something like sortVolumeServer

**Review by remyleone — APPROVED** [review]

**Pull request merged**